tracer/src/Datadog.Trace/Activity/Handlers/DefaultActivityHandler.cs

```diff
@@ -0,0 +1,289 @@
+            {
+                if (activeSpan is not null)
+                {
+                    // If this is the first activity (no parent) and we already have an active span
+                    // or the span was started after the parent activity so we use the span as a parent
+
+                    // We ensure the activity follows the same TraceId as the span
+                    // And marks the ParentId the current spanId
+
+                    if (activity.Parent is null || activity.Parent.StartTimeUtc < activeSpan.StartTime.UtcDateTime)
+                    {
+                        // TraceId
+                        w3cActivity.TraceId = string.IsNullOrWhiteSpace(activeSpan.Context.RawTraceId) ?
+                                                  activeSpan.TraceId.ToString("x32") : activeSpan.Context.RawTraceId;
+
+                        // SpanId
+                        w3cActivity.ParentSpanId = string.IsNullOrWhiteSpace(activeSpan.Context.RawSpanId) ?
+                                                       activeSpan.SpanId.ToString("x16") : activeSpan.Context.RawSpanId;
+
+                        // We clear internals Id and ParentId values to force recalculation.
+                        w3cActivity.RawId = null;
+                        w3cActivity.RawParentId = null;
+
+                        // Avoid recalculation of the traceId.
+                        traceId = activeSpan.TraceId;
+                    }
+                }
+
+                // We convert the activity traceId and spanId to use it in the
+                // Datadog span creation.
+                traceId ??= Convert.ToUInt64(w3cActivity.TraceId.Substring(16), 16);
+                spanId = Convert.ToUInt64(w3cActivity.SpanId, 16);
+                rawTraceId = w3cActivity.TraceId;
+                rawSpanId = w3cActivity.SpanId;
+            }
+
+            try
+            {
+                Log.Debug($"DefaultActivityHandler.ActivityStarted: [Source={sourceName}, Id={activity.Id}, RootId={activity.RootId}, OperationName={{OperationName}}, StartTimeUtc={{StartTimeUtc}}, Duration={{Duration}}]", activity.OperationName, activity.StartTimeUtc, activity.Duration);
+
+                foreach (var ignoreSourceName in IgnoreOperationNamesStartingWith)
+                {
+                    if (activity.OperationName?.StartsWith(ignoreSourceName) == true)
+                    {
+                        if (activity is IW3CActivity w3cAct && activeSpan is not null)
+                        {
+                            // If we ignore the activity and there's an existing active span
+                            // We modify the activity spanId with the one in the span
+                            // The reason for that is in case this ignored activity is used
+                            // for propagation then the current active span will appear as parentId
+                            // in the context propagation, and we will keep the entire trace.
```

Comment:
Yes, and we already do the same in the `IgnoreActivityHandler` :D

Reply:
Ok, then forgive me, but wouldn't it make sense for the `IgnoreOperationNamesStartingWith` being in `IgnoreActivityHandler` too? I realise they're slightly different in that they're "start with" instead of "matching", but not sure if they need to be in the default handler, and they seem to make more sense to be in the ignore one? 🤔

Reply:
Well, each handler is attached by `source` name, in this case we want to create span from the source name, that's why we attached the default handler, but there's a couple of `OperationNames` (eg: HttpClient.Out` ) that we want to ignore. That's why we support ignoring by operation name in the default handler. I think I can move this logic to the Ignore handler and just call the method from the default one. WDYT?

Reply:
Ah, I see 🤔 

> I think I can move this logic to the Ignore handler and just call the method from the default one. WDYT?

Yeah, if it's easy maybe makes sense, but not required IMO

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/2446/commits/ec5588480ac085208ab6e5e269c85ed714f07f7b